src/universities/resolvers/faculties.resolver.ts

```
@@ -14,6 +16,16 @@ import { FacultyDegreesLoader } from '../loaders/faculty-degrees.loader';
 export class FacultiesResolver {
   constructor(private readonly universitiesService: UniversitiesService) {}
 
+  @Query(_returns => Faculty)
+  async getFacultyById(
+    @Args({ name: 'facultyId', type: () => ID })
+    facultyId: string,
+  ): Promise<Faculty> {
+    const res = await this.universitiesService.getFacultyById(facultyId);
+    if (res) return res;
+    else throw new NotFoundException();
```

Using `NotFoundException` probably doesn't make sense here: it's specific to HTTP, and won't do something sensible in the context of GraphQL. (It will just be relayed to the client as a crash.)

The query result should probably be made nullable (by adding `{ nullable: true }` to the `@Query`).

---

We should keep it consistent for what we currently have in this case. I think doing something like `      throw new ServerError('Failed to get faculty by id', err);` (what we currently do for the queries) is the best here. In the future we can look at doing nullable query results, but my preference is to just keep this as consistent as possible.

---

Hmm, speaking of consistency, we should probably rename this to `getFaculty` too?

---

I didn't actually see anywhere in the code where you guys resolve cases where an object is not found. Is that something that is done on the FE? 

Removing null from the return type on ```   async getFacultyById(facultyId: string): Promise<Faculty | null> {  ```  alows me to return the result and not have to worry about whether the faculty object was found or not.
Should I keep the null return type from there or thow the ServerError('Failed to get faculty by id', err) ?

---

@Nghondzweni: Checking for the `null` and raising an exception should be good, yeah, like `QueryDataService.getQuery`.

We probably don't have great handling of missing results right now, but it's not really something that's part of the main app flow right now (the faculty data should remain be relatively constant in practice), so that's okay for now. 

So, summary:

- [ ] Rename `getFacultyById` → `getFaculty` (consistency)
- [ ] Raise `ServerError` instead of returning `null` (consistency)
- [ ] Update test to pass

---

All changes requested on feedback summary completed.